src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -821,6 +821,28 @@ CHIPQueueLevel0::~CHIPQueueLevel0() {
   }
 }
 
+std::vector<ze_event_handle_t> CHIPQueueLevel0::addDependenciesQueueSync(
+    std::shared_ptr<chipstar::Event> TargetEvent) {
+  auto EventsToWaitOn = getSyncQueuesLastEvents();
+  // Every event in EventsToWaitOn should have a dependency on MemCopyEvent so
```

Comment:
Cannot we get "last events" updated after the call and before we enter the next loop to add the next dep?